Dashboard/Connectors/Serial/SerialConnector.cs

```diff
@@ -48,8 +80,7 @@ private void ParseMessage(string message)
         if (!MyRegex().IsMatch(message))
             throw new InvalidOperationException("Invalid message format received", new Exception(message));
 
-        // All messages start with the format "#ID=<ID>|UTC=<TIME>|<MSG>", so we split the message by the '|', and remove the first 2 elements
-        var split = message.Split('|')[2..];
+        var split = message.Substring(1).Split('|');
```

Comment:
1. It's to grab the ID and UTC time while trimming the first # off the ID.
2. I believe that was something I put in place when I brought up the buffer issue, you had asked to change the check to check for < 2 rather than != 2 as a stopgap. I can change it back to != 2.
3.  I changed it from using the Add function so it handles double up key-value pairs since we weren't checking for double ups. Currently I have it using the latest value of the key-value pair.